src/main/java/org/cactoos/iterator/Paged.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.cactoos.Func;
+import org.cactoos.Scalar;
+import org.cactoos.func.UncheckedFunc;
+import org.cactoos.scalar.Sticky;
+import org.cactoos.scalar.Unchecked;
+
+/**
+ * Paged iterator.
+ * Elements will continue to be provided so long as {@code next} produces
+ * non-empty iterators.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @param <I> Subtype of item iterator
+ * @param <X> Type of item
+ * @since 0.49
+ */
+public final class Paged<I extends Iterator<X>, X> implements Iterator<X> {
+
+    /**
+     * Current element.
+     */
+    private final AtomicReference<Unchecked<I>> current;
```

Comment:
@baudoliver7 just for clarity, I believe it's best to store here the `Scalar` directly, and wrap in unchecked only in `hasNext` and `next`

Reply:
@victornoel I see but the reason why it's done like this is for the need of storing the current value. By wanting to respect the principle of immutability, we need to store somewhere the current item after calling `hasNext`. And AtomicReference coupled to Unchecked serve the purpose of it.

Reply:
@baudoliver7 I think this discussion is moot because of https://github.com/yegor256/cactoos/pull/1536#discussion_r560400853, so I resolve this discussion :)